flow/connectors/clickhouse/normalize.go

```diff
@@ -67,65 +64,140 @@ func (c *ClickhouseConnector) SetupNormalizedTable(
+			clickhouseType, err = colType.ToDWHColumnType(protos.DBType_CLICKHOUSE)
+			if err != nil {
+				return "", fmt.Errorf("error while converting column type to clickhouse type: %w", err)
+			}
 		}
 
 		if colType == qvalue.QValueKindNumeric {
 			precision, scale := datatypes.GetNumericTypeForWarehouse(column.TypeModifier, datatypes.ClickHouseNumericCompatibility{})
 			if column.Nullable {
-				stmtBuilder.WriteString(fmt.Sprintf("`%s` Nullable(DECIMAL(%d, %d)), ", colName, precision, scale))
+				stmtBuilder.WriteString(fmt.Sprintf("`%s` Nullable(DECIMAL(%d, %d)), ", dstColName, precision, scale))
 			} else {
-				stmtBuilder.WriteString(fmt.Sprintf("`%s` DECIMAL(%d, %d), ", colName, precision, scale))
+				stmtBuilder.WriteString(fmt.Sprintf("`%s` DECIMAL(%d, %d), ", dstColName, precision, scale))
 			}
 		} else if tableSchema.NullableEnabled && column.Nullable && !colType.IsArray() {
-			stmtBuilder.WriteString(fmt.Sprintf("`%s` Nullable(%s), ", colName, clickhouseType))
+			stmtBuilder.WriteString(fmt.Sprintf("`%s` Nullable(%s), ", dstColName, clickhouseType))
 		} else {
-			stmtBuilder.WriteString(fmt.Sprintf("`%s` %s, ", colName, clickhouseType))
+			stmtBuilder.WriteString(fmt.Sprintf("`%s` %s, ", dstColName, clickhouseType))
 		}
 	}
 	// TODO support soft delete
 	// synced at column will be added to all normalized tables
-	if syncedAtColName != "" {
-		colName := strings.ToLower(syncedAtColName)
-		stmtBuilder.WriteString(fmt.Sprintf("`%s` %s, ", colName, "DateTime64(9) DEFAULT now64()"))
+	if config.SyncedAtColName != "" {
+		colName := strings.ToLower(config.SyncedAtColName)
+		stmtBuilder.WriteString(fmt.Sprintf("`%s` DateTime64(9) DEFAULT now64(), ", colName))
+	}
+
+	var engine string
+	if tableMapping == nil || tableMapping.Engine == protos.TableEngine_CH_ENGINE_REPLACING_MERGE_TREE {
+		engine = "ReplacingMergeTree"
+	} else {
+		engine = "MergeTree"
 	}
 
 	// add sign and version columns
 	stmtBuilder.WriteString(fmt.Sprintf(
-		"`%s` %s, `%s` %s) ENGINE = ReplacingMergeTree(`%s`)",
-		signColName, signColType, versionColName, versionColType, versionColName))
+		"`%s` %s, `%s` %s) ENGINE = %s(`%s`)",
+		signColName, signColType, versionColName, versionColType, engine, versionColName))
 
+	var pkeyStr string
 	pkeys := tableSchema.PrimaryKeyColumns
 	if len(pkeys) > 0 {
+		if len(colNameMap) > 0 {
+			pkeys = slices.Clone(pkeys)
+			for idx, pk := range pkeys {
+				pkeys[idx] = getColName(colNameMap, pk)
+			}
+		}
 		pkeyStr := strings.Join(pkeys, ",")
 
 		stmtBuilder.WriteString("PRIMARY KEY (")
 		stmtBuilder.WriteString(pkeyStr)
 		stmtBuilder.WriteString(") ")
+	}
+
+	orderby := make([]*protos.ColumnSetting, len(tableMapping.Columns))
+	for _, col := range tableMapping.Columns {
+		if col.Ordering > 0 && !slices.Contains(pkeys, getColName(colNameMap, col.SourceName)) {
+			orderby = append(orderby, col)
+		}
+	}
+	slices.SortStableFunc(orderby, func(a *protos.ColumnSetting, b *protos.ColumnSetting) int {
```

Comment:
do we want to allow multiple columns with the same `.Ordering`?

Reply:
Not going to go out the way to not handle it. SortStableFunc gives it a consistent result